tests/e2e_test/module_deletion_test.go

```diff
@@ -0,0 +1,157 @@
+			Should(Succeed())
+		By("verifying remote kyma is ready")
+		Eventually(CheckRemoteKymaCR).
+			WithContext(ctx).
+			WithArguments(remoteNamespace, []v1beta2.Module{}, runtimeClient, v1beta2.StateReady).
+			Should(Succeed())
+	})
+
+	It("Should enable Template Operator and Kyma should result in Ready status", func() {
+		By("Enabling Template Operator")
+		Eventually(EnableModule).
+			WithContext(ctx).
+			WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", "regular", runtimeClient).
+			Should(Succeed())
+		By("Checking state of kyma")
+		Eventually(CheckKymaIsInState).
+			WithContext(ctx).
+			WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateReady).
+			Should(Succeed())
+	})
+
+	It("Should disable Template Operator with existing blocking finalizer "+
+		"and Kyma should result in Processing status", func() {
+		By("Inserting finalizer into Sample CR")
+		Eventually(SetFinalizer).
+			WithContext(ctx).
+			WithArguments("sample-yaml", "kyma-system", "operator.kyma-project.io/v1alpha1", "Sample",
+				[]string{"sample.kyma-project.io/finalizer", "blocking-finalizer"}, runtimeClient).
+			Should(Succeed())
+		By("Disabling Template Operator")
+		Eventually(DisableModule).
+			WithContext(ctx).
+			WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", runtimeClient).
+			Should(Succeed())
+		By("Checking state of kyma set to `Processing`")
+		Eventually(CheckKymaIsInState).
+			WithContext(ctx).
+			WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateProcessing).
+			Should(Succeed())
+		By("Sample CR should not be removed")
+		Eventually(CheckIfExists).
+			WithContext(ctx).
+			WithArguments("sample-yaml", "kyma-system", "operator.kyma-project.io/v1alpha1", "Sample",
+				runtimeClient).
+			Should(Succeed())
+		By("Checking template-operator deployment still exists")
+		Eventually(CheckIfExists).
+			WithContext(ctx).
+			WithArguments("template-operator-controller-manager", "template-operator-system", "apps/v1",
+				"Deployment", runtimeClient).
+			Should(Succeed())
+		By("Checking if it is possible to update Sample CR")
+		Eventually(UpdateSampleCRSpec).
+			WithContext(ctx).
+			WithArguments("sample-yaml", "kyma-system", "newValue", runtimeClient).
+			Should(Succeed())
+	})
+
+	It("Should be able to re-enable module under deletion", func() {
+		By("Re-Enabling Template Operator")
+		Eventually(EnableModule).
+			WithContext(ctx).
+			WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", "regular", runtimeClient).
+			Should(Succeed())
+		By("Removing all finalizers from sample CR")
+		Eventually(SetFinalizer).
+			WithContext(ctx).
+			WithArguments("sample-yaml", "kyma-system", "operator.kyma-project.io/v1alpha1", "Sample",
+				[]string{}, runtimeClient).
+			Should(Succeed())
+		By("New Sample CR should exists")
+		Eventually(CheckIfExists).
```

Comment:
By this check, you can't assume this Sample CR is new one, so additionally, we need to check the deletiontimestamp not exists.